### PR TITLE
Stats: refactor to use notices hooks

### LINKED ISSUE #77932: Paid Stats: Use notices hooks instead to show/hide notices
The context behind this is [the notices was changed to call a remote API instead of a local request](https://github.com/Automattic/jetpack/pull/31261).

### PR DESCRIPTION
**Pull request by kangzj**
Related to #77932 

## Proposed Changes

The PR refactored Stats Dashboard notice to leverage notices API rather than relying on notices in the initial state. It also removes redux related code not in use anymore.

This is a cleanup PR for [the Paid Stats Banner task](https://github.com/Automattic/wp-calypso/issues/78653).

## Testing Instructions

* Create a new JN site
* Connect the site to Jetpack
* Follow option 2 for Odyssey testing (sandbox widgets.wp.com)
* Ensure Opt Out Notice is shown and could be dismissed
* After 3~4 refreshes
* Ensure Feedback notice is shown and could be dismissed

image

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-notices/index.jsx [hunk 1]
```diff
@@ -1,96 +1,119 @@
...
-										/>
-									),
-								},
-							}
-						) }
-					</NoticeBanner>
-				</div>
-			) }
-		</>
+		! noticeDismissed && (
```
Comment (dognose24) [inline client/my-sites/stats/stats-notices/index.jsx]:
```suggestion
		! noticeDismissed && isOdysseyStats && (
```

Could we check the isOdysseyStats earlier?

**Review by dognose24 — APPROVED** [review]
This change looks reasonable and works as described on the JN site with Jetpack Bleeding Edge. I just left a nit that is not a blokcer.

File: client/my-sites/stats/stats-notices/index.jsx (hunk 1 above)
Comment (kangzj) [reply client/my-sites/stats/stats-notices/index.jsx]: The reason I put it this way is that we are going to add notice for Calypso Stats too, and `isOdyssey` would then be moved down 😃

**Review by jsnmoon — APPROVED** [review]
Looks great and works a treat 🚀

**Comment by kangzj** [comment]
@jsnmoon @dognose24 sorry friends, I just figured we needed to be compatible with old Odyssey, so the removal of old code shouldn't happen just yet.

I recovered the old mechanism and refactored the code to share the notice templates between new/old. I would appreciate another review please. 🙇

File: client/my-sites/stats/stats-notices/feedback-notice.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { StatsNoticeProps, FeedbackNoticeBodyProps } from './types';
+
+export const FeedbackNoticeBody = ( {
+	onTakeSurveyClick,
+	onRemindMeLaterClick,
+	onNoticeDismiss,
+}: FeedbackNoticeBodyProps ) => {
+	const translate = useTranslate();
+
+	return (
```
Comment (kangzj) [inline client/my-sites/stats/stats-notices/feedback-notice.tsx:15]: moved from old `index.jsx`

File: client/my-sites/stats/stats-notices/opt-out-notice.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import useNoticeVisibilityMutation from 'calypso/my-sites/stats/hooks/use-notice-visibility-mutation';
+import useNoticeVisibilityQuery from 'calypso/my-sites/stats/hooks/use-notice-visibility-query';
+import { StatsNoticeProps, NoticeBodyProps } from './types';
+
+export const OptOutNoticeBody = ( { onNoticeDismiss }: NoticeBodyProps ) => {
+	const translate = useTranslate();
+
+	return (
```
Comment (kangzj) [inline client/my-sites/stats/stats-notices/opt-out-notice.tsx:11]: moved from old `index.jsx`

File: client/my-sites/stats/stats-notices/legacy-notices.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { updateStatsNoticeStatusDirect } from 'calypso/state/sites/notices/actions';
+import hasNewStatsFeedbackNotice from 'calypso/state/sites/selectors/has-new-stats-feedback-notice';
+import hasOptOutNewStatsNotice from 'calypso/state/sites/selectors/has-opt-out-new-stats-notice';
+import { FeedbackNoticeBody } from './feedback-notice';
+import { OptOutNoticeBody } from './opt-out-notice';
+
+export default function LegacyStatsNotices( { siteId } ) {
+	const dispatch = useDispatch();
```
Comment (kangzj) [inline client/my-sites/stats/stats-notices/legacy-notices.jsx:9]: moved from old `index.jsx`

**Review by jsnmoon — APPROVED** [review]
Good call on backward compatibility. These changes appear reasonable.

I assume we won't be able to deprecate the legacy notices as long as customers continue using the older Jetpack plugin versions?

**Comment by kangzj** [comment]
> I assume we won't be able to deprecate the legacy notices as long as customers continue using the older Jetpack plugin versions?

IMO the notices are fine to be removed after a couple of releases as they are not core functionality.

**Pull request merged**